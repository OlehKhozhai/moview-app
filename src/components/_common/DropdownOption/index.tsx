import React from 'react';
import cn from 'classnames';

import styles from './styles.module.scss';

type DropdownOptionProps = {
  option: string;
  className?: string;
  onOptionClick: (value: string) => void;
};

const DropdownOption: React.FC<DropdownOptionProps> = ({ option, className, onOptionClick }) => {
  const handleOptionClick = (e: React.MouseEvent) => {
    e?.stopPropagation();
    onOptionClick(option);
  };

  return (
    <li className={cn(styles.root, className)} onClick={handleOptionClick}>
      {option}
    </li>
  );
};

export default React.memo(DropdownOption);
