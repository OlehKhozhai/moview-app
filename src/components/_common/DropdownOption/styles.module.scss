@import 'assets/styles/mixins/colors';

.root {
  padding: 10px 15px;
  text-transform: capitalize;

  &:hover {
    background-color: $primary-dark-bg;
    border-radius: 5px;
  }
}
