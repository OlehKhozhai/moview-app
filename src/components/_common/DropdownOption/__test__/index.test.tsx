import React from 'react';
import { create } from 'react-test-renderer';

import DropdownOption from '../index';

describe('DropdownOption component', () => {
  test('should render DropdownOption component by default', () => {
    const component = create(
      <DropdownOption option="Test option" onOptionClick={() => undefined} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should trigger onOptionClick method', () => {
    const handleOnOptionClick = jest.fn();
    const component = create(
      <DropdownOption option="Test option" onOptionClick={handleOnOptionClick} />
    );

    component.root.findByType('li').props.onClick();

    expect(handleOnOptionClick).toHaveBeenCalledWith('Test option');
  });
});
