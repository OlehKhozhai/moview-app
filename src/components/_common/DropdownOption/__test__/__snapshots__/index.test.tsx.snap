// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownOption component should render DropdownOption component by default 1`] = `
<li
  className="root"
  onClick={[Function]}
>
  Test option
</li>
`;
